app/ante/vesting.go

```diff
@@ -0,0 +1,101 @@
+package ante
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+	vestingtypes "github.com/tharsis/evmos/x/vesting/types"
+)
+
+type VestingDelegationDecorator struct {
+	ak evmtypes.AccountKeeper
+}
+
+func NewVestingDelegationDecorator(ak evmtypes.AccountKeeper) VestingDelegationDecorator {
+	return VestingDelegationDecorator{
+		ak: ak,
+	}
+}
+
+func (vdd VestingDelegationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	// check if the tx contains a staking delegation and error if the tokens are still locked or the bond amount is greater than the tokens already vested
+	for _, msg := range tx.GetMsgs() {
+		for _, addr := range msg.GetSigners() {
+
+			delegateMsg, isDelegation := msg.(*stakingtypes.MsgDelegate)
+			if !isDelegation {
+				continue
+			}
+
+			acc := vdd.ak.GetAccount(ctx, addr)
+			if acc == nil {
+				// TODO error msg
+				return ctx, fmt.Errorf("account doesnt exists")
+			}
+
+			clawbackAccount, isPeriodicVesting := acc.(*vestingtypes.ClawbackVestingAccount)
+			if !isPeriodicVesting {
+				// continue to next decorator as this logic only applies to vesting
+				return next(ctx, tx, simulate)
+			}
+
+			// error if bond amount is > vested tokens
+			coins := clawbackAccount.GetVestedOnly(ctx.BlockHeader().Time)
+			vested := coins.AmountOf(stakingtypes.DefaultParams().BondDenom)
```

Comment:
this will always be `"stake"`, we need to get the params from the staking keeper

Reply:
To solve this it looks like we need to add the `StakingKeeper` to the ante `HandlerOptions`. Right now this is defined on ethermint. Can we also define them on evmos?

Reply:
yeah we can define them on Evmos too 👍

Reply:
nice, added them on evmos!